src/Kiota.Builder/Writers/Go/CodeMethodWriter.cs

```diff
@@ -262,7 +262,8 @@ private void WriteRequestBuilderBody(CodeClass parentClass, CodeMethod codeEleme
                     writer.WriteLine($"{defaultValueReference} := {propWithDefault.DefaultValue};");
                     defaultValueReference = $"&{defaultValueReference}";    
                 }
-                writer.WriteLine($"m.Set{propWithDefault.SymbolName.ToFirstCharacterUpperCase()}({defaultValueReference});");
+                var setterName = propWithDefault.SetterFromCurrentOrBaseType?.Name.ToFirstCharacterUpperCase() ?? $"Set{propWithDefault.SymbolName.ToFirstCharacterUpperCase()}";
```

Comment:
Seems like `SetterFromCurrentOrBaseType` and `Setter` can be used interchangeably. Not sure when to use which.
Perhaps `SetterFromCurrentOrBaseType` should be `SetterFromBaseType`? and `Setter` to refer to the current class?

Reply:
I got the impression that one would always want to call `SetterFromCurrentOrBaseType` since it covers either case?

Reply:
Sure but it has additional costs (conditions & Co). I was thinking that leaving both leads to a choice for the writer which can use the simplest one when needed. There might be cases where you only want the one from the current class for some reason.